lib/src/string_scanner.dart

```diff
@@ -120,6 +139,45 @@ class StringScanner {
     _fail(name);
   }
 
+  /// Consumes a single Unicode code unit and returns it.
+  ///
+  /// This works like [readChar], except that it automatically handles UTF-16
+  /// surrogate pairs. Specifically, if the next character is a high surrogate,
+  /// it consumes it as well as the following character and returns the resolved
+  /// Unicode code point.
+  ///
+  /// If the scanner is looking at an unpaired surrogate, this returns the high
+  /// surrogate as-is.
+  int readCodePoint() {
+    final first = readChar();
```

Comment:
This is calling `readChar` which is public and virtual and overridden in `LineScanner` to update the line/column.

That means that `LineScanner.readCodePoint` calls `_adjustLineAndColumn` three times for the same `readCodePoint` call.

You want to call the non-overridden `readChar` (which would mean copying it to a private method and let the public `readChar` call the private one), or just do as in `scanChar` and access the string directly.

Reply:
Updated to simply not override `readCodePoint()` in `LineScanner`, since this will automatically do the correct thing without the extra `_adjustLineAndColumn()`.